pinot-core/src/main/java/org/apache/pinot/core/query/pruner/ColumnValueSegmentPruner.java

```diff
@@ -239,6 +239,49 @@ private boolean pruneRangePredicate(IndexSegment segment, RangePredicate rangePr
     return false;
   }
 
+  /**
+   * For IN predicate, prune the segment based on:
+   * <ul>
+   *   <li>Column min/max value</li>
+   * </ul>
+   */
```

Comment:
(nit) please consider adding more details to javadoc highlighting when the pruning won't happen if number of values is greater than threshold.